eth/executionclient/execution_client.go

```diff
@@ -117,6 +124,66 @@ func (ec *ExecutionClient) FetchHistoricalLogs(ctx context.Context, fromBlock ui
 	return logs, lastBlock, nil
 }
 
+type FetchedLogs struct {
+	logs      []ethtypes.Log
+	lastBlock uint64
+	err       error
+}
+
+// Calls FilterLogs multiple times and batches results to avoid fetching enormous amount of events
+func (ec *ExecutionClient) batchedFilterLogs(ctx context.Context, client *ethclient.Client, query ethereum.FilterQuery) <-chan FetchedLogs {
+
+	logsChan := make(chan FetchedLogs)
+
+	go func() {
+		defer close(logsChan)
+
+		q := math.Floor(float64((query.ToBlock.Uint64() - query.FromBlock.Uint64()) / ec.logBatchSize))
```

Comment:
Why do we need `q` to be `float64`? Can we just do 
```go
q := query.ToBlock.Uint64() - query.FromBlock.Uint64()) / ec.logBatchSize
```
instead?

Reply:
I wanted to be sure that any uin64 division will not be rounded to upper bound. Looks like Go uint division should give a floor as well. Lets go with 
```
q := query.ToBlock.Uint64() - query.FromBlock.Uint64()) / ec.logBatchSize
```